spec/support/shared_examples/declarative_updates.rb

```diff
@@ -350,3 +358,41 @@ def generate_new_value(attribute_to_change:)
     end
   end
 end
+
+RSpec.shared_examples "a declarative conditional touch model", :with_touches do |when_changing: [], on_event: %i[update], timestamp_attribute: :updated_at, target_optional: true|
```

Comment:
I think we're missing a lot of the tests in the other shared context here; this doesn't seem to account for create/destroy, target_optional etc? Instead of having a separate shared context would we be able to work it into the main one? so something like:

```
it_behaves_like "a conditional touch model", when_changing: %i[started_on finished_on], timestamp_attribute: :api_updated_at, conditional_method: :touch_teacher?
```

Then in the existing shared context default the `conditional_method` to `nil` and mock it out to return `true` if its specified, then we should just need one extra test that mocks it to `false` and checks the update doesn't happen?